### ISSUE TITLE
Can't select Grouped list using Space/Enter

### ISSUE THREAD

**Issue by cnamal-msft** [microsoft/fluentui#19771]
Hello 👋

### Describe the issue:

[For Grouped List](https://fabricweb.z5.web.core.windows.net/oufr/8.34.0/#/examples/detailslist/largegrouped), when the focus is on the group name, hitting space/enter to (un)select the whole sub category doesn't work.

![image](https://user-images.githubusercontent.com/53378748/133109796-8da3c71f-4484-471f-8c5c-c903710699ae.png)

Looking at the history, it used to work until [7.140.0](https://fabricweb.z5.web.core.windows.net/oufr/7.140.0/#/examples/detailslist/largegrouped) and was broken in [7.141.0](https://fabricweb.z5.web.core.windows.net/oufr/7.141.0/#/examples/detailslist/largegrouped)

### Environment Information

- **Package version(s)**: last version (8.34.0)
- **Browser and OS versions**: Edge

Impacted products: AI Builder in PowerApps

**Comment by gouttierre** [comment]
@cnamal-msft - Thanks for filing this issue with us. I see the repro, hitting space/enter on the header row seems to work where the sub category doesn't work. To set expectation our team will have a look at the issue but only once capacity allows. 
  
@GeoffCoxMSFT  -  Would you be able to confirm if this is a regression, or if this behavior is an issue?
